extractor/src/main/java/org/schabi/newpipe/extractor/stream/AudioStream.java

```diff
@@ -4,109 +4,416 @@
+        return super.equalStats(cmp) && cmp instanceof AudioStream
+                && averageBitrate == ((AudioStream) cmp).averageBitrate;
     }
 
     /**
-     * Get the average bitrate
-     * @return the average bitrate or -1
+     * Get the average bitrate of the stream.
+     *
+     * @return the average bitrate or {@link #UNKNOWN_BITRATE} if it is unknown
      */
     public int getAverageBitrate() {
-        return average_bitrate;
+        return averageBitrate;
     }
 
+    /**
+     * Get the itag identifier of the stream.
+     *
+     * <p>
+     * Always equals to {@link #ITAG_NOT_AVAILABLE_OR_NOT_APPLICABLE} for other streams than the
+     * ones of the YouTube service.
+     * </p>
+     *
+     * @return the number of the {@link ItagItem} passed in the constructor of the audio stream.
+     */
     public int getItag() {
         return itag;
     }
 
+    /**
+     * Get the bitrate of the stream.
+     *
+     * @return the bitrate set from the {@link ItagItem} passed in the constructor of the stream.
+     */
     public int getBitrate() {
         return bitrate;
     }
 
+    /**
+     * Get the initialization start of the stream.
+     *
+     * @return the initialization start value set from the {@link ItagItem} passed in the
+     * constructor of the stream.
+     */
     public int getInitStart() {
         return initStart;
     }
 
+    /**
+     * Get the initialization end of the stream.
+     *
+     * @return the initialization end value set from the {@link ItagItem} passed in the constructor
+     * of the stream.
+     */
     public int getInitEnd() {
         return initEnd;
     }
 
+    /**
+     * Get the index start of the stream.
+     *
+     * @return the index start value set from the {@link ItagItem} passed in the constructor of the
+     * stream.
+     */
     public int getIndexStart() {
         return indexStart;
     }
 
+    /**
+     * Get the index end of the stream.
+     *
+     * @return the index end value set from the {@link ItagItem} passed in the constructor of the
+     * stream.
+     */
     public int getIndexEnd() {
         return indexEnd;
     }
 
+    /**
+     * Get the quality of the stream.
+     *
+     * @return the quality label set from the {@link ItagItem} passed in the constructor of the
+     * stream.
+     */
     public String getQuality() {
         return quality;
     }
 
+    /**
+     * Get the codec of the stream.
+     *
+     * @return the codec set from the {@link ItagItem} passed in the constructor of the stream.
+     */
     public String getCodec() {
         return codec;
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @Nullable
+    public ItagItem getItagItem() {
+        return itagItem;
+    }
+
+    @Override
+    public boolean equals(final Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
```

Comment:
Why are these two check not in the super method in `Stream`?
Same question for the other `equals` methods which extend `Stream`.

Reply:
They are also in the super method, I used what Stypox suggested [here](https://github.com/TeamNewPipe/NewPipeExtractor/pull/663#discussion_r790264754) (so what IDEA did).